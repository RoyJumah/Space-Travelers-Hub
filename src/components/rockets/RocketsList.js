import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import Rocket from './Rocket';
import { fetchRockets, selectAllRockets, selectState } from '../../redux/rockets/rocketsSlice';
import styles from './css/RocketsList.module.scss';

const RocketsList = () => {
  const dispatch = useDispatch();
  const status = useSelector(selectState);
  const contents = useSelector((state) => state.rockets.contents);
  useEffect(() => {
    if (contents.length === 0 && status === 'idle') {
      dispatch(fetchRockets());
    }
  });
  const rockets = useSelector(selectAllRockets);
  return (
    <ul className={styles.rocketsList}>
      {rockets.map((rocket) => (
        <Rocket
          key={rocket.id}
          rocket={rocket}
        />
      ))}
    </ul>
  );
};

export default RocketsList;
