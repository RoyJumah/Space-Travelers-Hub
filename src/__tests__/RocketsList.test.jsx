import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import RocketsList from '../components/rockets/RocketsList';
import { Provider } from 'react-redux';
import store from '../redux/ConfigureStore';

describe('Display RocketsList component', () => {
  test('it matches snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <RocketsList />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })

  test('should return  RocketsLIst Component', () => {
    render(
      <Provider store={store}>
        <RocketsList />
      </Provider>
    );
  })
})
