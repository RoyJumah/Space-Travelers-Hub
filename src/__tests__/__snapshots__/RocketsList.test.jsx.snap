// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display RocketsList component it matches snapshot 1`] = `
<ul
  className="rocketsList"
/>
`;
